src/components/select/select.ts

```diff
@@ -0,0 +1,377 @@
+  public static override styles = styles;
+  private searchTerm = '';
+  private lastKeyTime = Date.now();
+
+  @queryAssignedElements({ flatten: true, selector: 'igc-select-item' })
+  protected override items!: Array<IgcSelectItemComponent>;
+
+  @queryAssignedElements({ flatten: true, selector: 'igc-select-group' })
+  protected override groups!: Array<IgcSelectGroupComponent>;
+
+  @queryAssignedElements({ slot: 'helper-text' })
+  protected helperText!: Array<HTMLElement>;
+
+  @queryAssignedElements({ slot: 'suffix' })
+  protected inputSuffix!: Array<HTMLElement>;
+
+  @queryAssignedElements({ slot: 'prefix' })
+  protected inputPrefix!: Array<HTMLElement>;
+
+  @state()
+  protected override selectedItem!: IgcSelectItemComponent | null;
+
+  @query('igc-input')
+  protected override target!: IgcInputComponent;
+
+  /** The value attribute of the control. */
+  @property({ reflect: false, type: String })
+  public value!: string | undefined;
+
+  /** The name attribute of the control. */
+  @property()
+  public name!: string;
+
+  /** The disabled attribute of the control. */
+  @property({ reflect: true, type: Boolean })
+  public disabled = false;
+
+  /** The required attribute of the control. */
+  @property({ reflect: true, type: Boolean })
+  public required = false;
+
+  /** The invalid attribute of the control. */
+  @property({ reflect: true, type: Boolean })
+  public invalid = false;
+
+  /** The outlined attribute of the control. */
+  @property({ reflect: true, type: Boolean })
+  public outlined = false;
+
+  /** The autofocus attribute of the control. */
+  @property({ type: Boolean })
+  public override autofocus!: boolean;
+
+  /** The label attribute of the control. */
+  @property({ type: String })
+  public label!: string;
+
+  /** The placeholder attribute of the control. */
+  @property({ type: String })
+  public placeholder!: string;
+
+  /** Whether the dropdown's width should be the same as the target's one. */
+  @property({ type: Boolean, attribute: 'same-width' })
+  public override sameWidth = true;
+
+  /** The direction attribute of the control. */
+  @property({ reflect: true })
+  public override dir: 'ltr' | 'rtl' | 'auto' = 'auto';
+
+  constructor() {
+    super();
+    this.size = 'medium';
+  }
+
+  /** Sets focus on the component. */
+  public override focus(options?: FocusOptions) {
+    this.target.focus(options);
+  }
+
+  /** Removes focus from the component. */
+  public override blur() {
+    this.target.blur();
+  }
+
+  /** Checks the validity of the control. */
+  public reportValidity() {
+    this.invalid = this.required && !this.value;
+    if (this.invalid) this.focus();
+    return !this.invalid;
+  }
+
+  public override async firstUpdated() {
+    super.firstUpdated();
+    await this.updateComplete;
+
+    const selectedItem = this.allItems.find((i) => i.selected) ?? null;
+    if (selectedItem) this.value = selectedItem.value;
+    else this.updateSelected();
```

Comment:
If-else without any brackets, that's a new thing. Can't say I'm a fan of the readability.

Also if there's a selectedItem it should hit the `updateValue` watch below, no? Which in turn can (will?) trigger `updateSelected` since that's on a watch too?

Reply:
If there's a selected item it will hit, yes. I will refactor this. This wasn't possible before the dropdown refactoring. The only case that should be handled is when there's no `selectedItem` and `value` has been initialized on the select component.